metadata/metadata.go

```diff
@@ -272,6 +273,7 @@ func (wrapper SearchWrapper) Set(id ResourceID, res Resource) error {
 		Name:    id.Name,
 		Type:    id.Type.String(),
 		Variant: id.Variant,
+		Tags: 	 id.Tags,
```

Comment:
Hey @ahmadnazeri,
`Set` method is getting its content from `ResourceID` struct. But adding `Tags` to it (which is of `[]string` type) breaks the backward compatibility:

1. using maps with `ResrourceID` which fails since `map` can't compare `[]string` types.
2. will have to pass Tags whereever `ResourceID` is called. 
3. for functions like [ResourceLocation](https://github.com/featureform/featureform/blob/1b6004810e9fb7861dc2317309c5d2dfe7800c37/serving/serving.go#L386), will have to use GetTags() that will go into [ResourceID](https://github.com/featureform/featureform/blob/1b6004810e9fb7861dc2317309c5d2dfe7800c37/serving/serving.go#L424) 

So one way I can think of making it backward compatible is:
- For point 1,  create a custom mapping function that handles comparing []string types
- Point 2 and 3 will change lots of files so test coverage will be essential to know if everything is working.

Let me know if this seems the right way to handle this or creating another struct with tags is a better way to  implement this.

Reply:
Is it the schema in the search index thats breaking or just the change to the struct that would break something?

Reply:
Line 281 (just below in this page) has a map function:
`type LocalResourceLookup map[ResourceID]Resource` 

Since `id.Tags` has type `[]String` (line 268, code block shown below), `ResourceID` has the same tags type which will trigger an error on line 281 because maps can't compare []string types inherently. So to handle such issue I should create a custom map, I guess. 
```
func (wrapper SearchWrapper) Set(id ResourceID, res Resource) error {
	if err := wrapper.ResourceLookup.Set(id, res); err != nil {
		return err
	}
	doc := search.ResourceDoc{
		Name:    id.Name,
		Type:    id.Type.String(),
		Variant: id.Variant,
		Tags: 	 id.Tags,
	}
	return wrapper.Searcher.Upsert(doc)
}
```